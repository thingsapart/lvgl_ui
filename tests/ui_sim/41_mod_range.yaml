# TICKS: 3
- type: data-binding
  state:
    - count_up: 0.0
    - count_down: 10.0
  actions:
    - step:
        count_up: { range: }
        count_down: { range: [0, 10, -2] }
  schedule:
    - { tick: 1, action: step }
    - { tick: 2, action: step }
    - { tick: 3, action: step }
