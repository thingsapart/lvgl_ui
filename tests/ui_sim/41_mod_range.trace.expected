--- UI-Sim Trace Start ---
STATE_SET: count_up = 0.000 (old: null)
STATE_SET: count_down = 10.000 (old: null)
NOTIFY: count_up = 0.000
NOTIFY: count_down = 10.000

--- TICK 1 ---
ACTION: step value=null
STATE_SET: count_up = 1.000 (old: 0.000)
STATE_SET: count_down = 8.000 (old: 10.000)
NOTIFY: count_up = 1.000
NOTIFY: count_down = 8.000

--- TICK 2 ---
ACTION: step value=null
STATE_SET: count_up = 2.000 (old: 1.000)
STATE_SET: count_down = 6.000 (old: 8.000)
NOTIFY: count_up = 2.000
NOTIFY: count_down = 6.000

--- TICK 3 ---
ACTION: step value=null
STATE_SET: count_up = 0.000 (old: 2.000)
STATE_SET: count_down = 10.000 (old: 6.000)
NOTIFY: count_up = 0.000
NOTIFY: count_down = 10.000

--- UI-Sim Trace End ---
